contracts/src/Teleporter/upgrades/TeleporterRegistry.sol

```diff
@@ -0,0 +1,74 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import "../../WarpProtocolRegistry.sol";
+import "../ITeleporterMessenger.sol";
+
+/**
+ * @dev TeleporterRegistry contract is a {WarpProtocolRegistry} and provides an upgrade
+ * mechanism for teleporter messenger contracts.
+ */
+contract TeleporterRegistry is WarpProtocolRegistry {
+    mapping(address => uint256) internal _addressToVersion;
+
+    error DuplicateProtocolAddress();
+
+    constructor(
+        uint256[] memory initialVersions,
+        address[] memory initialProtocolAddresses
+    ) WarpProtocolRegistry(initialVersions, initialProtocolAddresses) {}
+
+    /**
+     * @dev Gets the {ITeleporterMessenger} contract of the given `version`.
+     */
+    function getTeleporter(
+        uint256 version
+    ) external view returns (ITeleporterMessenger) {
+        return ITeleporterMessenger(_getVersionToAddress(version));
+    }
+
+    /**
+     * @dev Gets the latest {ITeleporterMessenger} contract.
+     */
+    function getLatestTeleporter()
+        external
+        view
+        returns (ITeleporterMessenger)
+    {
+        return ITeleporterMessenger(_getVersionToAddress(_latestVersion));
+    }
+
+    /**
+     * @dev Gets the version of the given `protocolAddress`.
+     * If `protocolAddress` is not a valid protocol address, returns 0.
+     */
+    function getAddressToVersion(
```

Comment:
Nit: Could we rename this something along the lines of `getVersionFromAddress`? The fact that it starts with "getAddress" currently but is actually used to get a version was confusing to me for a minute.

We could make possibly make "get" method names symmetric then by changing `getTeleporter` to something like `getAddressFromVersion`

Reply:
renamed functions